### PR TITLE
Pass focus() to Input

### PR DESCRIPTION
**Pull request by kevashcraft**
Added a function to pass the focus command to the input element.

### CODE REVIEW

File: paper-input-behavior.html [hunk 1]
```diff
@@ -237,6 +237,13 @@
       return this.$.input;
     },
 
+    /**
+     * Passes focus() to the input element
+     */
+    focus: function() {
+      this.$.input.focus();
```
Comment (notwaldorf) [inline paper-input-behavior.html]: this should be `this.$.inputElement.focus()`

File: demo/index.html [hunk 2]
```diff
@@ -65,6 +65,13 @@
       <button onclick="validate()">Validate!</button>
     </div>
 
+    <h4>Focus</h4>
+    <div class="vertical-section">
+      <paper-input label="textarea label" id="inputForFocus"></paper-input>
```
Comment (notwaldorf) [inline demo/index.html]: I think `textarea` doesn't belong in this label :)

**Comment by notwaldorf** [comment]
(+@cdata, our focus expert)

@kevashcraft I know this is a one line change, but it's actually really contentious. `focus` is a native method, which has a specific behaviour, which you're overriding. For example, calling `focus()` on an element without a tabindex doesn't actually focus that element. However, in this specific case, all `paper-input`s are inherently focusable (because their `iron-inputs` are), and setting a `tabindex`on a `paper-input` leads to a double-focus scenario (shift-tabbing out of a focused `paper-input` changes the tab from the `iron-input` to the `paper-input` container and looks weird).

I would be ok if you renamed the method to something like `focusInput`, and did the same thing (plus the comments I left on the code). I would also really like to see a test of this behaviour, not just in the demo (also, for the `paper-textarea`) 😊

Super sorry about all this. Focus is actually really hard to get right 😢

**Comment by barakhub** [comment]
@notwaldorf, Sadly, Adding `focusInput` is not going to do the work.
Check out the following example: http://poly-play.appspot.com/C5FZY24iJmI
When the page opens, click on the 'Open modal' button and when the dialog opens hit shift+tab.

You'll see that the focus moves outside of the modal dialog to the 'Aleft 1' button. Hit space to see that it's really active.

The reason is here: https://github.com/PolymerElements/paper-dialog-behavior/blob/master/paper-dialog-behavior.html#L211
Once the focus moves outside of the dialog, the code in `PaperDialogBehavior._onFocus` tries to set it back to the last element with focus within the dialog, but if that's a `paper-input`, this fails (and this kind of change would fix that).

Generally, `paper-input` should "feel" like a built-in componnet and properly handle `focus`. Maybe it's better not to override `focus`, but instead handle the `focus` event.

**Comment by notwaldorf** [comment]
Closing this PR, since the fix will be addressed by #186

**Pull request closed without merging**